packages/markdown-editor/src/MarkdownEditor/plugins/ToolbarPlugin/ToolbarPlugin.tsx

```diff
@@ -375,6 +382,35 @@ export const ToolbarPlugin = ({ isDisabled, externalActions, toolbarPluginProps
     }
   }, [activeEditor, isLink]);
 
+  const toggleMarkdown = useCallback(() => {
+    setShowMarkdown((showMarkdown) => !showMarkdown);
+
+    editor.update(() => {
+      const root = $getRoot();
+      const firstChild = root.getFirstChild();
+
+      if ($isCodeNode(firstChild) && firstChild.getLanguage() === 'markdown') {
+        $convertFromMarkdownString(firstChild.getTextContent(), PLAYGROUND_TRANSFORMERS);
+      } else {
+        const markdown = $convertToMarkdownString(PLAYGROUND_TRANSFORMERS);
+        root.clear().append($createCodeNode('markdown').append($createTextNode(markdown)));
+      }
+      root.selectEnd();
+    });
+  }, [editor]);
+
+  const markdownToggleButton = (
+    <ToolbarIcon
+      icon={<IconMarkdown />}
+      onClick={toggleMarkdown}
+      isActive={showMarkdown}
+      isDisabled={isDisabled}
+      aria-label="Toggle Markdown on and off"
+      title="Toggle Markdown on and off"
+      data-testid="markdown-toggle-button"
```

Comment:
- aria-label is needed because this button has no label at all.
- title is used to show the "label" on hover, so people can hover the button and know what it does - but I can drop if it is decided to do so.
- getByRole returns 2 items. The other one is the Markdown option on the dropdown options.